tests/core/test_variable.py

```diff
@@ -0,0 +1,588 @@
+        self.assertAlmostEqual(w.value, 6.0)
+
+        #test creation with repeated attributes
+        w = DepVar({'t':t, 'u':u}, v=v, x=u)
+        self.assertCountEqual(w.params, ('t','u','v','x'))
+        self.assertDictEqual({p:v for p,v in w.depends}, {'t':t,'u':u,'v':v,'x':u})
+        self.assertAlmostEqual(w.value, 8.0)
+
+        #test creation with multiple vardicts
+        w = DepVar({'t':t, 'u':u}, {'v':v})
+        self.assertCountEqual(w.params, ('t','u','v'))
+        self.assertDictEqual({p:v for p,v in w.depends}, {'t':t,'u':u,'v':v})
+        self.assertAlmostEqual(w.value, 6.0)
+
+        #test invalid creation with no attributes
+        with self.assertRaises(AttributeError):
+            w = DepVar()
+
+    def test_dependency_graph(self):
+        """Test dependency_graph method."""
+        t = relentless.DesignVariable(value=1.0)
+        u = relentless.DesignVariable(value=2.0)
+        v = relentless.DesignVariable(value=3.0)
+
+        #test 1 level of dependency
+        w = DepVar(t=t, u=u, v=v)
+        g = w.dependency_graph()
+        self.assertCountEqual(g.nodes, [t,u,v,w])
+        self.assertCountEqual(g.edges, [(w,t),(w,u),(w,v)])
+        self.assertEqual(g.edges[w,t]['params'], ['t'])
+        self.assertEqual(g.edges[w,u]['params'], ['u'])
+        self.assertEqual(g.edges[w,v]['params'], ['v'])
+
+        #test 2 levels of dependency
+        x = DepVar(t=t, v=v)
+        y = DepVar(w=w, x=x)
+        g = y.dependency_graph()
+        self.assertCountEqual(g.nodes, [t,u,v,w,x,y])
+        self.assertCountEqual(g.edges, [(w,t),(w,u),(w,v),(x,t),(x,v),(y,w),(y,x)])
+        self.assertEqual(g.edges[w,t]['params'], ['t'])
+        self.assertEqual(g.edges[w,u]['params'], ['u'])
+        self.assertEqual(g.edges[w,v]['params'], ['v'])
+        self.assertEqual(g.edges[x,t]['params'], ['t'])
+        self.assertEqual(g.edges[x,v]['params'], ['v'])
+        self.assertEqual(g.edges[y,w]['params'], ['w'])
+        self.assertEqual(g.edges[y,x]['params'], ['x'])
+
+        #test more complex dependency
+        z = DepVar(t=t, w=w, y=y)
+        g = z.dependency_graph()
+        self.assertCountEqual(g.nodes, [t,u,v,w,x,y,z])
+        self.assertCountEqual(g.edges, [(w,t),(w,u),(w,v),(x,t),(x,v),(y,w),(y,x),
+                                        (z,t),(z,w),(z,y)])
+
+        #test 'multiple' dependency on same object
+        q = DepVar(t=t, u=t, v=t)
+        g = q.dependency_graph()
+        self.assertCountEqual(g.nodes, [t,q])
+        self.assertCountEqual(g.edges, [(q,t)])
+        self.assertCountEqual(g.edges[q,t]['params'], ['t','u','v'])
+
+        q = DepVar(t=t, u=t, v=v, w=v)
+        g = q.dependency_graph()
+        self.assertCountEqual(g.nodes, [t,v,q])
+        self.assertCountEqual(g.edges, [(q,t),(q,v)])
+        self.assertCountEqual(g.edges[q,t]['params'], ['t','u'])
+        self.assertCountEqual(g.edges[q,v]['params'], ['v','w'])
+
+        #test circular dependencies
```

Comment:
For all of these tests, could you call the method `DependentVariable._assert_acyclic(g)` and confirm that it throws? Maybe you should also call it once above on a graph free from cycles.